proposals/0000-splitting-out-stable-interfaces-from-th.rst

```diff
@@ -0,0 +1,203 @@
+Users who wish to continue to use ``template-haskell`` may continue to do so, and the interfaces will continue to be exposed in both the new packages and the old.
+
+By focusing on smaller subsets of the API we also make it much easier to be compatible with multiple versions of GHC.
+This is an important property for any stable package as it allows a user to upgrade their dependencies independently of GHC.
+We plan to implement this by create compatibility shims using ``CPP`` or ``PatternSynonyms``.
+
+.. _advantages:
+Benefits of splitting out ``template-haskell-lift`` and ``template-haskell-quasiquote``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Publishing ``template-haskell-lift`` and ``template-haskell-quasiquote`` will be beneficial both for GHC and the ecosystem.
+
+The biggest benefit is that library authors who are just deriving or using ``Lift`` instances or just exposing ``Quasiquoter``\s no longer need to depend on the entirety of ``template-haskell``.
+This can help avoid the sorts of dependency bounds propagation problems identified in the `GHC.X.Hackage proposal <https://github.com/bgamari/tech-proposals/blob/ghc-x-hackage/proposals/001-ghc-x-hackage.md>`_.
+
+There is a more subtle benefit for the ``template-haskell`` package. Currently the wide usage of ``Lift`` instances greatly limits the possible dependencies of ``template-haskell``.
+For instance, ``template-haskell`` cannot depend on ``containers`` or ``filepath``, since these libraries depend on ``template-haskell``.
+But if these packages switch to depending on our new packages, then ``template-haskell`` could depend on them.
+Currently ``template-haskell`` must vendor a small portion of ``filepath`` and ``containers``, and that would no longer be necessary.
+
+Many boot packages depend on ``template-haskell``, but all of them only depend on it for the parts of the interface exposed by ``template-haskell-lift`` and ``template-haskell-quasiquote``.
+If we can convince their maintainers to depend on these packages instead, then GHC would no longer (transitively) depend on ``template-haskell``.
+This makes it possible for packages to depend on the ``ghc`` library at the same time as a version of ``template-haskell`` different to the one bundled with that GHC.
+
+
+Proposed Change Specification
+-----------------------------
+No changes to the language or the compiler are required for this proposal.
+
+Proposed Library Change Specification
+-------------------------------------
+
+We propose to publish two new libraries: ``template-haskell-lift`` and ``template-haskell-quasiquote``.
+These will be shipped with GHC.
+They will also be buildable from Hackage.
+They will be buildable with at a *minimum* the last 3 versions of GHC.
+The current version of the libraries are compatible with GHC 8.10 and later.
+
```

Comment:
Mention what are dependencies are.
Mention that the library is reinstallable but shipped with GHC ala containers, bytestring, etc, but not baked in like ghc-internal.